packages/driver/src/dom/coordinates.ts

```diff
@@ -7,12 +7,26 @@ const getElementAtPointFromViewport = (doc, x, y) => {
   return $elements.elementFromPoint(doc, x, y)
 }
 
-const isAutIframe = (win) => {
+const isAUTFrame = (win) => {
   const parent = win.parent
 
   // https://github.com/cypress-io/cypress/issues/6412
   // ensure the parent is a Window before checking prop
-  return $window.isWindow(parent) && !$elements.getNativeProp(parent, 'frameElement')
+  if (!$window.isWindow(parent)) {
+    return false
+  }
+
+  try {
+    // window.frameElement only exists on iframe windows, so if it doesn't
+    // exist on parent, it must be the top frame, and `win` is the AUT
+    return !$elements.getNativeProp(parent, 'frameElement')
+  } catch (err) {
+    // if the AUT is cross-origin, accessing parent.frameElement will throw
```

Comment:
Do we need to add a check to this error to ensure it's a cross-origin error before returning true?

Reply:
Addressed in https://github.com/cypress-io/cypress/pull/21137/commits/2f7d248302c72f066112ecc2e44108d361a74912